File: CS/RemoveNewButtonInLookups.Module.Web/WebDeactivateNewActionInLookupsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp.Web.Templates;
using DevExpress.ExpressApp.Editors;

namespace RemoveNewButtonInLookups.Module.Win {
    public class WebDeactivateNewActionInLookupsController : DeactivateNewActionInLookupsController {
        protected override bool IsLookupTemplate() {
            var page = Frame.Template as BaseXafPage;
            if (page != null) 
                return page.TemplateContent is ILookupPopupFrameTemplate;
            return false;
        }
    }
}

File: VB/RemoveNewButtonInLookups.Module.Web/WebDeactivateNewActionInLookupsController.vb
Imports System.Text
Imports DevExpress.ExpressApp.Web.Templates
Imports DevExpress.ExpressApp.Editors

Namespace RemoveNewButtonInLookups.Module.Win
    Public Class WebDeactivateNewActionInLookupsController
        Inherits DeactivateNewActionInLookupsController

        Protected Overrides Function IsLookupTemplate() As Boolean
            Dim page = TryCast(Frame.Template, BaseXafPage)
            If page IsNot Nothing Then
                Return TypeOf page.TemplateContent Is ILookupPopupFrameTemplate
            End If
            Return False
        End Function
    End Class
End Namespace
